ai-service/app/ml/recommendation/service.py

```
@@ -0,0 +1,270 @@
+
+        # For now, use a simple frequency-based approach
+        # In production, this would use the full bipartite graph
+        user_property_ids = {b.get("propertyId") for b in bookings if b.get("propertyId")}
+
+        if not user_property_ids:
+            return None
+
+        # Simple heuristic: properties that share attributes with booked ones score higher
+        scores = {}
+        for pid in candidate_ids:
+            p = property_map_global.get(pid) if hasattr(self, '_property_map') else None
```

Ruff (0.16.2)

[error] Undefined name `property_map_global`

(F821)